client/client.go

```diff
@@ -200,6 +206,17 @@ func (e ConnectionError) Unwrap() error {
 	return e.Err
 }
 
+type InternalError struct{ Err error }
```

Comment:
I like this usage of the new errors.Is etc a lot, but I'm not sure on the name, InternalError is very vague, and it's also not clear which component the error is "internal to", maybe instead this could be ClientInternalError to make it clear this type of error is only for internal errors on the client side and not i.e. Internal Server Error 500 from the snapd

Reply:
I see your point, I renamed it so it's clearer